node/pkg/vaa/structs_test.go

```diff
@@ -488,8 +489,157 @@ func TestVerifySignaturesFuzz(t *testing.T) {
+			amount:         725000000,
+			errString:      "",
+		},
+		{label: "unsupported payload type",
+			vaa:       "01000000000100e424aef95296cb0f2185f351086c7c0b9cd031d1288f0537d04ab20d5fc709416224b2bd9a8010a81988aa9cb38b378eb915f88b67e32a765928d948dc02077e00000102584a8d000000020000000000000000000000000290fb167208af455bb137780163b7b7a9a10c16000000000000000f0f02000000000000000000000000000000000000000000000000000000002b369f40000000000000000000000000ddb64fe46a91d46ee29420539fc25fd07c5fea3e000221c175fcd8e3a19fe2e0deae96534f0f4e6a896f4df0e3ec5345fe27ac3f63f000010000000000000000000000000000000000000000000000000000000000000000",
+			errString: "unsupported payload type",
+		},
+		{label: "buffer too short",
+			vaa:       "01000000000100e424aef95296cb0f2185f351086c7c0b9cd031d1288f0537d04ab20d5fc709416224b2bd9a8010a81988aa9cb38b378eb915f88b67e32a765928d948dc02077e00000102584a8d000000020000000000000000000000000290fb167208af455bb137780163b7b7a9a10c16000000000000000f0f01",
+			errString: "buffer too short",
+		},
+		{label: "empty string",
+			vaa:       "",
+			errString: "buffer too short",
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(string(tc.label), func(t *testing.T) {
+			data, err := hex.DecodeString(tc.vaa)
+			assert.Nil(t, err)
+
+			vaa, err := Unmarshal(data)
+			assert.Nil(t, err)
+			assert.NotNil(t, vaa)
+
+			if len(tc.errString) == 0 {
+				expectedEmitterAddr, err := StringToAddress(tc.emitterAddr)
+				assert.Nil(t, err)
+
+				expectedTokenAddr, err := StringToAddress(tc.tokenAddr)
+				assert.Nil(t, err)
+
+				expectedToAddr, err := StringToAddress(tc.toAddr)
+				assert.Nil(t, err)
+
+				expectedAmount := big.NewInt(tc.amount)
+
+				assert.Equal(t, tc.emitterChainId, vaa.EmitterChain)
+				assert.Equal(t, expectedEmitterAddr, vaa.EmitterAddress)
+				assert.Equal(t, 133, len(vaa.Payload))
+
+				payload, err := DecodeTransferPayloadHdr(vaa.Payload)
+				assert.Nil(t, err)
+				assert.Equal(t, tc.payloadType, payload.Type)
+				assert.Equal(t, tc.tokenChainId, payload.TokenChainID)
+				assert.Equal(t, expectedTokenAddr, payload.TokenAddress)
+				assert.Equal(t, tc.toChainId, payload.ToChainID)
+				assert.Equal(t, expectedToAddr, payload.ToAddress)
+				assert.Equal(t, expectedAmount.Cmp(payload.Amount), 0)
+			} else {
+				_, err = DecodeTransferPayloadHdr(vaa.Payload)
+				assert.NotNil(t, err)
+				assert.Equal(t, tc.errString, err.Error())
+			}
+
+		})
+	}
+}
+
+func TestIsTransfer(t *testing.T) {
+	type Test struct {
+		label   string
+		payload []byte
+		result  bool
+	}
+
+	tests := []Test{
+		{label: "empty", payload: []byte{}, result: false},
+		{label: "non-valid payload", payload: []byte{0x0}, result: false},
+		{label: "payload 1", payload: []byte{0x1}, result: true},
+		{label: "payload 2", payload: []byte{0x2}, result: false},
+		{label: "payload 3", payload: []byte{0x3}, result: true},
+		{label: "payload 4", payload: []byte{0x4}, result: false},
```

Comment:
Would there be cases where we'd expect that a Wormhole message might be using a payload type and providing malicious inputs here to muck with the stats or do we only expect payloads at this point to be fully trusted?

I'm thinking if this contributes to a limit, can I forge arbitrary payloads to run the numbers up?  If so, how should/would we guard against that?